### ISSUE TITLE
missing support for web components

### ISSUE THREAD

**Issue by mattlohkamp** [javan/details-element-polyfill#60]
As written, this polyfill only addresses `<detail>` support in the light DOM - i.e. `<style>`s added to `document.head` will not apply to elements encapsulated in the shadow DOM that comes standard with web components.

Just off the top of my head, allowing the polyfill functionality to be manually initialized with context passed in as an argument, might be a fairly direct fix - make the master function an export, and the implementation looks something like this:

```javascript
import dePolyfill from 'details-element-polyfill';
window.customElement.define('my-element', class MyElement extends HTMLElement {
 this.attachShadow({mode:'open');
 dePolyfill(this.shadowRoot);
});
```

**Comment by javan** [comment]
Out of curiosity, are there browsers with Shadow DOM  support that don't support `<details>`?

**Comment by mattlohkamp** [comment]
I don't think so. I need to support IE11 so I'm knee-deep in polyfill land though. 😓
